### PR TITLE
Add record picker with variables

### PR DESCRIPTION
**Pull request by thomtrp**
- Add update actions
- Create a folder for workflow actions
- Add a SingleRecordPicker with variables handler

https://github.com/user-attachments/assets/9fd57ce1-1b8d-424a-8aa1-69468d684fa1

### CODE REVIEW

File: packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  objectNameSingular: string;
+};
+
+export const WorkflowSingleRecordPicker = ({
+  label,
+  defaultValue,
+  objectNameSingular,
+  onChange,
+}: WorkflowSingleRecordPickerProps) => {
+  const [draftValue, setDraftValue] = useState<
```
Comment (martmull) [inline packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx:89]: naming suggestion 
```suggestion
  const [rawValue, setRawValue] = useState<
```

File: packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        onRemove={handleUnlinkVariable}
+      />
+    ) : draftValue && draftValue.type === 'static' && selectedRecord ? (
+      <StyledRecordChip
+        record={selectedRecord}
+        objectNameSingular={objectNameSingular}
+      />
+    ) : (
+      <StyledPlaceholder>Select a {objectNameSingular}</StyledPlaceholder>
+    );
```
Comment (martmull) [inline packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx:186]: readability : if else if else please

File: packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    setDraftValue({
+      type: 'static',
+      value: '',
+    });
+    closeDropdown();
+
+    onChange('');
+  };
+
+  const Chip =
```
Comment (martmull) [inline packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx:171]: I would create a dedicated component for that (WorkflowVariableChip?)

File: packages/twenty-front/src/modules/workflow/workflow-actions/components/WorkflowEditActionFormServerlessFunction.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { useGetManyServerlessFunctions } from '@/settings/serverless-functions/hooks/useGetManyServerlessFunctions';
-import { WorkflowEditActionFormServerlessFunctionInner } from '@/workflow/components/WorkflowEditActionFormServerlessFunctionInner';
 import { WorkflowCodeAction } from '@/workflow/types/Workflow';
+import { WorkflowEditActionFormServerlessFunctionInner } from '@/workflow/workflow-actions/components/WorkflowEditActionFormServerlessFunctionInner';
```
Comment (martmull) [inline packages/twenty-front/src/modules/workflow/workflow-actions/components/WorkflowEditActionFormServerlessFunction.tsx:3]: can we undo this move please? I will remove this file in my pr and the merge conflict will be easier to fix without that move

**Comment by FelixMalfait** [comment]
🤩

File: packages/twenty-front/src/modules/workflow/workflow-actions/components/WorkflowEditActionFormServerlessFunction.tsx (hunk 4 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/workflow/workflow-actions/components/WorkflowEditActionFormServerlessFunction.tsx:3]: sure

File: packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx (hunk 2 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx:186]: 😂 you're right

File: packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx (hunk 1 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx:89]: I wanted to be consistent with the other FieldInput created by @Devessier. Why do you think this is better?

Comment (martmull) [reply packages/twenty-front/src/modules/workflow/components/WorkflowSingleRecordPicker.tsx:78]: raw seems more appropriate to me regarding draftValue usage. But it is not a mandatory change and consistency between components is more important. You can resolve

**Review by martmull — APPROVED** [review]
lgtm nice job!

**Pull request merged**